### PR TITLE
Add coverage badge to read-me doc

### PR DESCRIPTION
**Pull request by atifaziz**
# PR for issue #366

## What is being addressed

Coverage badge for the project.

## How is this addressed

Added to the read-me document of the project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 [![LoRa CI](https://github.com/Azure/iotedge-lorawan-starterkit/actions/workflows/ci.yaml/badge.svg?branch=master)](https://github.com/Azure/iotedge-lorawan-starterkit/actions/workflows/ci.yaml)
 [![LoRa CI](https://github.com/Azure/iotedge-lorawan-starterkit/actions/workflows/ci.yaml/badge.svg?branch=dev)](https://github.com/Azure/iotedge-lorawan-starterkit/actions/workflows/ci.yaml)
 [![Markdown](https://github.com//Azure/iotedge-lorawan-starterkit/actions/workflows/md-linter.yaml/badge.svg?branch=dev)](https://github.com/Azure/iotedge-lorawan-starterkit/actions/workflows/md-linter.yaml)
+[![codecov](https://codecov.io/gh/Azure/iotedge-lorawan-starterkit/branch/dev/graph/badge.svg?token=OTwrbFLcPS)](https://codecov.io/gh/Azure/iotedge-lorawan-starterkit)
```
Comment (danigian) [inline README.md:6]: What about this token? Is it a secret one?

Comment (atifaziz) [reply README.md]: > What about this token?

This was provided in the URL in the [badges page of the project](https://docs.codecov.com/docs/status-badges).

> Is it a secret one?

I don't know, to be honest. It doesn't seem to be documented. If I was to take a guess, I imagine it's for private repos, but this one is public. I tried the URL without the token and it works fine without it so I've removed it with efd1f28d8bc8213af4ee7aea6d805077f6af1519.

**Review by danigian — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (p-schuler) [reply README.md]: If this was a secret we would need to rotate it now.

Comment (atifaziz) [reply README.md]: I checked with Codecov support and am told it's not considered a secret. Quoting the reply:

> That is called the graph token, and it is good only for calls to the graph endpoint for badges and the charts. It's not needed for public repos, but it is for private ones.
>
> We don't consider is a secret, based on it's read-only nature to coverage numbers…

There is no UI for regenerating the token and it's not linked to the upload token (which can be regenerated).

Comment (atifaziz) [reply README.md]: While the token is no secret, I worked with Codecov support to generate a new one, just to err on the safe side.

**Pull request merged**